src/kernels/MIOpenBatchNormFwdTrainSpatial.cl

```diff
@@ -32,9 +32,9 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
-#ifdef __AMDGCN__
+/* #ifdef __AMDGCN__
```

Comment:
Good to me

Reply:
@muralinr @daniellowell 
This macro is implementation-defined so basically we shouldn't #define or #undef it.

I recommend defining another macro and using it, e.g.
```C++
#ifdef __AMDGCN__
#define USE_AMDGCN 1
#else
#define USE_AMDGCN 0
#endif
...
#if USE_AMDGCN
    local _FLOAT_ACCUM lcl_data_x[MIO_BN_LDS_SIZE];
    local _FLOAT_ACCUM lcl_data_y[MIO_BN_LDS_SIZE];
    lds_reduce2(&mean, &variance, (_FLOAT_ACCUM)INHW, lcl_data_x, lcl_data_y, lid);
#else
    local _FLOAT_ACCUM lcl_data_x[MIO_BN_LDSGCN_SIZE];
    local _FLOAT_ACCUM lcl_data_y[MIO_BN_LDSGCN_SIZE];
    gcn_reduce2(&mean, &variance, (_FLOAT_ACCUM)INHW, lcl_data_x, lcl_data_y, lid);
#endif
```
in reduction_functions.h:
```C++
#if USE_AMDGCN // was #ifdef ___AMDGCN__
...
```
in batchnorm_functions.h:
```C++
#ifndef MIO_BN_NODPP
#define MIO_BN_NODPP 0
#elif(MIO_BN_NODPP == 1 && MIO_BN_VARIANT != 0)
#undef USE_AMDGCN // was #undef __AMDGCN__
#define USE_AMDGCN 0 //
#endif
...
```

Reply:
[Informative] The previous review comment is just FYI. This PR is good to go.